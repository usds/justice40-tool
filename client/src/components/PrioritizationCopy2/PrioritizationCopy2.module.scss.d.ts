declare namespace PrioritizationCopy2Namespace {
    export interface IPrioritizationCopy2Scss {
      prioritizationCopy2Container: string;
    }
  }

declare const PrioritizationCopy2ScssModule: PrioritizationCopy2Namespace.IPrioritizationCopy2Scss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: PrioritizationCopy2Namespace.IPrioritizationCopy2Scss;
  };

  export = PrioritizationCopy2ScssModule;
