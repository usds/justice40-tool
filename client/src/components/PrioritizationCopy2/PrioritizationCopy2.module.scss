@use '../../styles/design-system.scss' as *;

.prioritizationCopy2Container{
    @include u-padding-top(2);
}
