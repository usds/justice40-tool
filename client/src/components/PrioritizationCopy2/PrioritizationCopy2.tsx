import React from 'react';

import * as styles from './PrioritizationCopy2.module.scss';

import * as EXPLORE_COPY from '../../data/copy/explore';

interface IPrioritizationCopy2 {
  totalCategoriesPrioritized: number
  totalBurdensPrioritized?: number
  isAdjacencyThreshMet: boolean,
  isAdjacencyLowIncome: boolean,
  tribalCountAK: number | null,
  tribalCountUS: null, // when this signal is supported add number type
  percentTractTribal: number | null
};

/**
 * This component returns the prioritzation copy.
 *
 * The truth table for this logic is captured in the shared doc called "Indicators for
 * UI and BE Signals". The Sheet name is "Disadv Copy p2 v5". This sheet is a copy of the
 * the full table in Sheet "Disadv Copy v4 p1", which is the full truth table in logical order.
 * The p2 v4 sheet copies the full truth table and sorts by the column 'second paragraph' to
 * find the appropriate logical grouping for the second paragraph.
 *
 * @param {number} totalCategoriesPrioritized
 * @param {number} totalBurdensPrioritized
 * @param {boolean} isAdjacencyThreshMet
 * @param {boolean} isAdjacencyLowIncome
 * @param {number | null} tribalCountAK
 * @param {number | null} tribalCountUS
 * @param {number | null} percentTractTribal
 * @return {JSX}
 */
const PrioritizationCopy2 =
   ({totalCategoriesPrioritized,
     isAdjacencyThreshMet,
     isAdjacencyLowIncome,
     tribalCountAK,
     tribalCountUS,
     percentTractTribal,
   }:IPrioritizationCopy2) => {
     let noStyles = false;
     let prioCopy2Rendered;

     // if 1
     if (
       (totalCategoriesPrioritized === 0 && (isAdjacencyThreshMet && isAdjacencyLowIncome)) ||
      (totalCategoriesPrioritized >= 1)
     ) {
       // if 1-1
       if (
         tribalCountAK === null &&
        (tribalCountUS !== null && tribalCountUS >= 1) &&
        (percentTractTribal !== null && percentTractTribal >= 1)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioPercAndNumPointsAlsoCopy(`${percentTractTribal}%`, tribalCountUS);
       // if 1-2
       } else if (
         tribalCountAK === null &&
        tribalCountUS === null &&
        (percentTractTribal !== null && percentTractTribal >= 1)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTCopy(`${percentTractTribal}%`, true);
       // if 1-3
       } else if (
         tribalCountAK === null &&
        (tribalCountUS !== null && tribalCountUS >= 1) &&
        (percentTractTribal !== null && percentTractTribal == 0)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioPercAndNumPointsAlsoCopy(`less than 1%`, tribalCountUS);
       // if 1-4
       } else if (
         tribalCountAK === null &&
        tribalCountUS === null &&
        (percentTractTribal !== null && percentTractTribal == 0)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTCopy(`less than 1%`, true);
       // if 1-5
       } else if (
         (tribalCountAK !== null && tribalCountAK >= 1) &&
        tribalCountUS === null &&
        percentTractTribal === null
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioANVCopy(tribalCountAK, true);
       // if 1-6
       } else if (
         (tribalCountAK !== null && tribalCountAK >= 1) &&
        (tribalCountUS !== null && tribalCountUS >= 1)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioAKUSCopy(tribalCountAK, tribalCountUS, true);
         // if 1-7
       } else if (
         !isAdjacencyThreshMet && isAdjacencyLowIncome &&
       (tribalCountAK !== null && tribalCountAK >= 1) &&
       (percentTractTribal !== null && percentTractTribal >= 1)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTCopy(`${percentTractTribal}%`, true);
         // if 1-8
       } else if (
         (totalCategoriesPrioritized >= 1) &&
         tribalCountAK == null &&
         (tribalCountUS !== null && tribalCountUS >= 1) &&
         percentTractTribal == null
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTPointsCopy(tribalCountUS, true);
         // if 1-9
       } else if (
         totalCategoriesPrioritized >= 1 &&
       (tribalCountAK !== null && tribalCountAK >= 1) &&
       (percentTractTribal !== null && percentTractTribal == 0)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTCopy(`less than 1%`, true);
         // if 1-10
       } else if (
         totalCategoriesPrioritized >= 1 &&
       (tribalCountAK !== null && tribalCountAK >= 1) &&
       (percentTractTribal !== null && percentTractTribal >= 1)
       ) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTCopy(`${percentTractTribal}%`, true);
       }
       // if 2
     } else if (
       totalCategoriesPrioritized === 0 &&
      isAdjacencyThreshMet && !isAdjacencyLowIncome &&
      tribalCountAK === null && tribalCountUS === null
     ) {
       // if 2-1
       if (percentTractTribal !== null && percentTractTribal == 0) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTCopy(`less than 1%`, false);
         // if 2-2
       } else if (percentTractTribal !== null && percentTractTribal >= 1) {
         prioCopy2Rendered = EXPLORE_COPY.getPrioFRTCopy(`${percentTractTribal}%`, false);
       }
       // if 3-1
     } else if (
       totalCategoriesPrioritized === 0 &&
      (isAdjacencyThreshMet && !isAdjacencyLowIncome) &&
      tribalCountAK === null &&
      (tribalCountUS !== null && tribalCountUS >= 1) &&
      (percentTractTribal !== null && percentTractTribal == 0)
     ) {
       prioCopy2Rendered = EXPLORE_COPY.getPrioFRTPointsCopy(tribalCountUS, false);
       // if 3-2
     } else if (
       totalCategoriesPrioritized === 0 &&
      (isAdjacencyThreshMet && !isAdjacencyLowIncome) &&
      tribalCountAK === null &&
      (tribalCountUS !== null && tribalCountUS >= 1) &&
      (percentTractTribal !== null && percentTractTribal >= 1)
     ) {
       prioCopy2Rendered = EXPLORE_COPY.getPrioFRTPointsCopy(tribalCountUS, false);
       // if 3-3
     } else if (
       (totalCategoriesPrioritized === 0 && !(isAdjacencyThreshMet && isAdjacencyLowIncome)) &&
      tribalCountAK === null &&
      (tribalCountUS !== null && tribalCountUS >= 1) &&
      (percentTractTribal !== null && percentTractTribal >= 0)
     ) {
       prioCopy2Rendered = EXPLORE_COPY.getPrioFRTPointsCopy(tribalCountUS, true);
       // if 4
     } else if (
       totalCategoriesPrioritized === 0 &&
      isAdjacencyThreshMet && !isAdjacencyLowIncome &&
      (tribalCountAK !== null && tribalCountAK >= 1)
     ) {
       prioCopy2Rendered = EXPLORE_COPY.getPrioANVCopy(tribalCountAK, false);
     } else {
       prioCopy2Rendered = <></>;
       noStyles = true;
     };

     return (
       <div className={noStyles ? '' : styles.prioritizationCopy2Container}>
         {prioCopy2Rendered}
       </div>
     );
   };

export default PrioritizationCopy2;
