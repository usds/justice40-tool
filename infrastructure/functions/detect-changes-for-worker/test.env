REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1