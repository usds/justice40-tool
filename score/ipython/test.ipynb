{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
