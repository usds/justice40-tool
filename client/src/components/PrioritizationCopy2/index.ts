import PrioritizationCopy2 from './PrioritizationCopy2';
export default PrioritizationCopy2;
