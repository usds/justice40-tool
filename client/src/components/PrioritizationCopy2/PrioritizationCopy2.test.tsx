import React from 'react';
import {render, screen} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import PrioritizationCopy2 from './PrioritizationCopy2';

describe('rendering of PrioritizationCopy2 Component', () => {
  const testCases = [
    {
      totalCategoriesPrioritized: 0,
      totalBurdensPrioritized: 0,
      isAdjacencyThreshMet: true,
      isAdjacencyLowIncome: true,
      tribalCountAK: null,
      tribalCountUS: null,
      percentTractTribal: 2,
      // eslint-disable-next-line max-len
      para1: `The lands of Federally Recognized Tribes that cover 2% of this tract are also considered disadvantaged.`,
    },
    {
      totalCategoriesPrioritized: 0,
      totalBurdensPrioritized: 1,
      isAdjacencyThreshMet: true,
      isAdjacencyLowIncome: true,
      tribalCountAK: null,
      tribalCountUS: 3,
      percentTractTribal: 4,
      // eslint-disable-next-line max-len
      para1: `The lands of Federally Recognized Tribes that cover 4% of this tract are also considered disadvantaged.`,
    },
  ];

  testCases.forEach((testCase) => {
    // eslint-disable-next-line max-len
    it(`checks if component renders ${testCase.para1} when totCats = ${testCase.totalCategoriesPrioritized}, totBurds = ${testCase.totalBurdensPrioritized}, isAdj = ${testCase.isAdjacencyThreshMet}, isAdjLI = ${testCase.isAdjacencyLowIncome}, tribal % = ${testCase.percentTractTribal},`, () => {
      const {asFragment} = render(
          <LocalizedComponent>
            <PrioritizationCopy2
              totalCategoriesPrioritized={testCase.totalCategoriesPrioritized}
              totalBurdensPrioritized={testCase.totalBurdensPrioritized}
              isAdjacencyThreshMet={testCase.isAdjacencyThreshMet}
              isAdjacencyLowIncome={testCase.isAdjacencyLowIncome}
              tribalCountAK={testCase.tribalCountAK}
              tribalCountUS={null}
              percentTractTribal={testCase.percentTractTribal}

            />
          </LocalizedComponent>,
      );
      expect(asFragment()).toMatchSnapshot();

      screen.getByText((content, element) => {
        return element?.tagName.toLowerCase() === 'div' && content.startsWith(testCase.para1);
      });
    });
  });
});
