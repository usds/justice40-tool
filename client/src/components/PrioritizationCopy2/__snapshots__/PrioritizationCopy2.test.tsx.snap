// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of PrioritizationCopy2 Component checks if component renders The lands of Federally Recognized Tribes that cover 2% of this tract are also considered disadvantaged. when totCats = 0, totBurds = 0, isAdj = true, isAdjLI = true, tribal % = 2, 1`] = `
<DocumentFragment>
  <div>
    The lands of Federally Recognized Tribes that cover 2% of this tract are also considered disadvantaged.
  </div>
</DocumentFragment>
`;

exports[`rendering of PrioritizationCopy2 Component checks if component renders The lands of Federally Recognized Tribes that cover 4% of this tract are also considered disadvantaged. when totCats = 0, totBurds = 1, isAdj = true, isAdjLI = true, tribal % = 4, 1`] = `
<DocumentFragment>
  <div>
    The lands of Federally Recognized Tribes that cover 4% of this tract are also considered disadvantaged.
  </div>
</DocumentFragment>
`;
